coral/src/app/features/topics/schema-request/TopicSchemaRequest.tsx

```diff
@@ -0,0 +1,166 @@
+// Since that is are important test cases, I opted for the bad practice rather in order to
+// be able to add this tests. I created na issue in github related to MonacoEditor testing already.
+function TopicSchemaRequest(props: TopicSchemaRequestProps) {
+  const { topicName } = props;
+
+  const navigate = useNavigate();
+  const form = useForm<TopicRequestFormSchema>({
+    schema: topicRequestFormSchema,
+    defaultValues: {
+      topicName: topicName,
+      schemafull: props.schemafullValueForTest || undefined,
+    },
+  });
+
+  const { data: environments, isLoading: environmentsIsLoading } = useQuery<
+    Environment[],
+    Error
+  >({
+    queryKey: ["schemaRegistryEnvironments"],
+    queryFn: () => getSchemaRegistryEnvironments(),
+  });
+  const schemaRequestMutation = useMutation(createSchemaRequest);
+
+  useEffect(() => {
+    if (window.msw !== undefined) {
+      mockGetSchemaRegistryEnvironments({
+        mswInstance: window.msw,
+        response: { data: mockedData },
+      });
+      mockCreateSchemaRequest({
+        mswInstance: window.msw,
+        response: { data: { status: "200 OK" } },
+      });
+    }
+  }, []);
+
+  if (schemaRequestMutation.isSuccess) {
+    const params = new URLSearchParams({
+      reqsType: "created",
+      schemaCreated: "true",
+    });
+    navigate(`/mySchemaRequests?${params.toString()}`);
+  }
+
+  function onSubmitForm(userInput: TopicRequestFormSchema) {
+    schemaRequestMutation.mutate(userInput);
+  }
+
+  function onCancel() {
+    form.reset();
+    navigate(-1);
+  }
+
+  return (
+    <Box style={{ maxWidth: 1200 }}>
+      {schemaRequestMutation.isError && (
+        <Box marginBottom={"l1"} role="alert">
+          <Alert
+            description={parseErrorMsg(schemaRequestMutation.error)}
+            type="warning"
+          ></Alert>
+        </Box>
+      )}
+      <Form
+        {...form}
+        ariaLabel={"Request a new schema"}
+        onSubmit={onSubmitForm}
+      >
+        <NativeSelect<TopicRequestFormSchema>
+          name={"topicName"}
+          labelText={"Topic name"}
+          defaultValue={topicName}
+          readOnly={true}
+          aria-readonly={true}
+        >
+          <option value={topicName}>{topicName}</option>
+        </NativeSelect>
+
+        {environmentsIsLoading && (
+          <div data-testid={"environments-select-loading"}>
+            <NativeSelect.Skeleton />
+          </div>
+        )}
+
+        {environments && (
+          <NativeSelect<TopicRequestFormSchema>
+            name={"environment"}
+            labelText={"Select environment"}
+            defaultValue={""}
+            required={true}
+          >
+            <option disabled value={""}>
+              please select
+            </option>
+            {environments.map((env, index) => {
+              return (
+                <option key={`${env.name}${index}`} value={env.id}>
```

Comment:
Lets use `env.id` as the key. That way we can drop usage of `index` from the iteration.